src/ctap/mod.rs

```diff
@@ -118,6 +121,17 @@ pub const ES256_CRED_PARAM: PublicKeyCredentialParameter = PublicKeyCredentialPa
     alg: SignatureAlgorithm::ES256,
 };
 
+// Helpers to perform CBOR read/write while respecting CTAP2 nesting limits.
+fn cbor_read(encoded_cbor: &[u8]) -> Result<cbor::Value, cbor::reader::DecoderError> {
+    cbor::reader::read_nested(encoded_cbor, Some(MAX_CBOR_NESTING_DEPTH))
```

Comment:
Can we map the error to `Err(Ctap2StatusCode::CTAP2_ERR_INVALID_CBOR)`? This would remove the conversion from `command.rs` line 54 (see other comment).

We could also decide to keep the error conversion function, and add one for `cbor::writer::EncoderError` too. But if we just convert the error in these respective functions, we have everything in one place.

Reply:
Good idea, that allows lots of places to just use `?` - done.